src/arviz_plots/plots/rootogramplot.py

```diff
@@ -0,0 +1,470 @@
+
+    # pp distribution group plotting logic
+    pp_distribution = process_group_variables_coords(
+        dt,
+        group=predictive_data_group,
+        var_names=(
+            None
+            if var_names is None
+            else [data_pairs.get(var_name, var_name) for var_name in var_names]
+        ),
+        filter_vars=filter_vars,
+        coords=coords,
+    )
+    # print(f"\npp_distribution = {pp_distribution}")
+
+    total_pp_samples = np.prod(
+        [pp_distribution.sizes[dim] for dim in sample_dims if dim in pp_distribution.dims]
+    )
+
+    # wrap plot collection with pp distribution
+    if plot_collection is None:
+        pc_kwargs.setdefault("col_wrap", 5)
+        pc_kwargs.setdefault(
+            "cols",
+            ["__variable__"]  # special variable to create one plot per variable
+            + list(facet_dims),  # making sure multiple plots are created for each facet dim
+        )
+        pc_kwargs["aes"] = pc_kwargs.get("aes", {}).copy()
+        pc_kwargs["aes"].setdefault("overlay", sample_dims)  # setting overlay dim
+        plot_collection = PlotCollection.wrap(
+            pp_distribution,
+            backend=backend,
+            **pc_kwargs,
+        )
+    # set reduce_dims by elimination
+    reduce_dims = [
+        dim for dim in pp_distribution.dims if dim not in set(facet_dims).union(set(sample_dims))
+    ]
+
+    if aes_map is None:
+        aes_map = {}
+    else:
+        aes_map = aes_map.copy()
+    aes_map.setdefault("predictive", plot_collection.aes_set)
+    if labeller is None:
+        labeller = BaseLabeller()
+
+    # setting plot_kwargs_dist defaults (for passing to internal plot_dist calls)
+    plot_kwargs_dist = {
+        key: False
+        for key in ("credible_interval", "point_estimate", "point_estimate_text", "title")
+    }
+    # if "remove_axis" in plot_kwargs:
+    plot_kwargs_dist["remove_axis"] = False  # plot_kwargs["remove_axis"]
+
+    # print(f"\n aes_map = {aes_map}")
+
+    # obs distribution calculated outside `if observed` since plotting predictive bars requires it
+    observed_data_group = "observed_data"
+    obs_distribution = process_group_variables_coords(
+        dt,
+        group=observed_data_group,
+        var_names=var_names,
+        filter_vars=filter_vars,
+        coords=coords,
+    )
+    # print(f"\n obs_distribution = {obs_distribution}")
+
+    # ---------(observed data)-----------
+    # observed data calculations are made outside of and before 'if observed' since predictive also
+    # depends on this computed data (number of bins and top of predictive bars for rootograms)
+
+    # use get_bins func from arviz-stats on observed data and then use those bins for
+    # computing histograms for predictive data as well
+    # WIP: currently only the bins for one variable (without any facetting) is retrieved and used
+    bins = array_stats.get_bins(obs_distribution["home_points"].values)
```

Comment:
Reminder for later: Use xarray aware `get_bins` function once available from `Arviz-Stats`